src/Microsoft.ML.Core/Data/MetadataUtils.cs

```diff
@@ -337,55 +307,24 @@ internal static void GetSlotNames(RoleMappedSchema schema, RoleMappedSchema.Colu
                 schema.Schema[list[0].Index].Metadata.GetValue(Kinds.SlotNames, ref slotNames);
         }
 
-        [BestFriend]
-        internal static bool HasKeyValues(this DataViewSchema.Column column, DataViewType type)
-        {
-            // False if type is not KeyType because GetKeyCount() returns 0.
-            ulong keyCount = type.GetKeyCount();
-            if (keyCount == 0)
-                return false;
-
-            var metaColumn = column.Metadata.Schema.GetColumnOrNull(Kinds.KeyValues);
-            return
-                metaColumn != null
-                && metaColumn.Value.Type is VectorType vectorType
-                && keyCount == (ulong)vectorType.Size
-                && vectorType.ItemType is TextDataViewType;
-        }
-
-        [BestFriend]
-        internal static bool HasKeyValues(this SchemaShape.Column col)
+        public static bool HasKeyValues(this SchemaShape.Column col)
```

Comment:
Just for my information - what's the difference between this method and the `public static bool HasKeyValues` in `SchemaAnnotationsExtensions`?